Mage.Sets/src/mage/cards/l/LegateLaniusCaesarsAce.java

```diff
@@ -0,0 +1,102 @@
+
+/**
+ * @author notgreat
+ */
+public final class LegateLaniusCaesarsAce extends CardImpl {
+
+    public LegateLaniusCaesarsAce(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{B}{R}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Decimate -- When Legate Lanius enters the battlefield, each opponent sacrifices a tenth of the creatures they control, rounded up.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new LegateLaniusCaesarsAceSacrificeEffect()).withFlavorWord("Decimate"));
+
+        // Whenever an opponent sacrifices a creature, put a +1/+1 counter on Legate Lanius.
+        this.addAbility(new SacrificePermanentTriggeredAbility(Zone.BATTLEFIELD,
+                new AddCountersSourceEffect(CounterType.P1P1.createInstance()),
+                StaticFilters.FILTER_PERMANENT_A_CREATURE, TargetController.OPPONENT, SetTargetPointer.NONE, false
+        ));
+    }
+
+    private LegateLaniusCaesarsAce(final LegateLaniusCaesarsAce card) {
+        super(card);
+    }
+
+    @Override
+    public LegateLaniusCaesarsAce copy() {
+        return new LegateLaniusCaesarsAce(this);
+    }
+}
+
+//Based on SacrificeAllEffect
+class LegateLaniusCaesarsAceSacrificeEffect extends OneShotEffect {
+    LegateLaniusCaesarsAceSacrificeEffect() {
+        super(Outcome.Sacrifice);
+        this.staticText = "each opponent sacrifices a tenth of the creatures they control, rounded up.";
+    }
+
+    private LegateLaniusCaesarsAceSacrificeEffect(final LegateLaniusCaesarsAceSacrificeEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LegateLaniusCaesarsAceSacrificeEffect copy() {
+        return new LegateLaniusCaesarsAceSacrificeEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Set<UUID> perms = new HashSet<>();
+        for (UUID playerId : game.getOpponents(source.getControllerId())) {
+            Player player = game.getPlayer(playerId);
+            if (player == null) {
+                continue;
+            }
+            // 1/10 rounded up
+            int num = (game.getBattlefield().count(StaticFilters.FILTER_PERMANENT_A_CREATURE,playerId,source, game)+9)/10;
+            int numTargets = Math.min(num, game.getBattlefield().count(TargetSacrifice.makeFilter(StaticFilters.FILTER_PERMANENT_A_CREATURE), player.getId(), source, game));
+            if (numTargets < 1) {
+                continue;
+            }
+            TargetSacrifice target = new TargetSacrifice(numTargets, StaticFilters.FILTER_PERMANENT_A_CREATURE);
+            while (!target.isChosen() && target.canChoose(playerId, source, game) && player.canRespond()) {
+                player.choose(Outcome.Sacrifice, target, source, game);
```

Comment:
AI can freeze here due wrong or cancelled choice.

Code protection for AI freeze:
- stop cycle on false choose (required for any "while" code);
- remove chosen value from a list for next choice by filter or cards list modification (recommended if possible).

So must add break here.

Reply:
This is currently exactly copied from the existing `SacrificeAllEffect`, and is also identical to `SacrificeEffect`'s. I'm not sure why this is a loop at all, honestly.

Reply:
I tried to remove the while loop when I refactored as part of #11587 but it caused a test failure, BarterInBloodTest I believe. I didn't investigate further why the choose method needs a loop to select multiple, seems not ideal. But it does serve a purpose.

Reply:
`BarterInBloodTest` uses AI targeting instead direct test commands and it fails on "while" remove. Looks like AI do not support multiple targets in choice dialogs (on resolve) and it works fine for cast/activate only (cause it uses game simulations for it).

Good code from human:
![shot_240402_070836](https://github.com/magefree/mage/assets/8344157/f1cb9817-31f7-4424-870c-ce7866c85112)

Bad code from computer:
![shot_240402_071218](https://github.com/magefree/mage/assets/8344157/2164132b-1e9a-4be4-b6fa-bb2a9293f8f0)

It's a more global problem with AI, not related to PR.